### PR TITLE
fetch-scm: init at v0.1.5

### PR DESCRIPTION
**Pull request by q60**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by q60** [comment]
note: github is evil about line endings

File: pkgs/tools/misc/fetch-scm/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, fetchFromGitHub, guile }:
```
Comment (figsoda) [inline pkgs/tools/misc/fetch-scm/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, guile }:
```

File: pkgs/tools/misc/fetch-scm/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+let
+  file = "fetch.scm";
+  pkg = "fetch-scm";
+in
+  stdenv.mkDerivation rec {
+    pname = pkg;
+    version = "v0.1.5";
```
Comment (figsoda) [inline pkgs/tools/misc/fetch-scm/default.nix:9]:
```suggestion
    version = "0.1.5";
```

File: pkgs/tools/misc/fetch-scm/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    src = fetchFromGitHub {
+      owner = "KikyTokamuro";
+      repo = file;
+      rev = version;
```
Comment (figsoda) [inline pkgs/tools/misc/fetch-scm/default.nix:14]:
```suggestion
      rev = "v${version}";
```

File: pkgs/tools/misc/fetch-scm/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{ stdenv, lib, fetchFromGitHub, guile }:
+
+let
+  file = "fetch.scm";
+  pkg = "fetch-scm";
```
Comment (figsoda) [inline pkgs/tools/misc/fetch-scm/default.nix:5]: please inline this and make sure the rest of the file doesn't have extra indentation

File: pkgs/tools/misc/fetch-scm/default.nix [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    };
+
+    dontBuild = true;
+    dontPatchShebangs = true;
+
+    buildInputs = [ guile ];
+
+    installPhase = ''
+      runHook preInstall
+      # strip shebang, because nix removes `--no-auto-compile`
```
Comment (figsoda) [inline pkgs/tools/misc/fetch-scm/default.nix:25]: the flag is not present in 0.1.5, I don't think nix removes it

File: pkgs/tools/misc/fetch-scm/default.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+      tail -n +5 ${file} > ${pkg}
+      # replace with new one
+      shebang="#!${guile.outPath}/bin/guile \\
+      --no-auto-compile -e main -s
+      !#
+      "
+      echo "$shebang$(cat ${pkg})" > ${pkg}
+      install -Dm755 -t $out/bin ${pkg}
```
Comment (figsoda) [inline pkgs/tools/misc/fetch-scm/default.nix:33]: i think we should keep the name of the executable unchanged

File: pkgs/tools/misc/fetch-scm/default.nix [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+      runHook postInstall
+    '';
+
+    meta = with lib; {
+      description = "System information fetcher written in GNU Guile Scheme";
+      homepage = "https://github.com/${src.owner}/${src.repo}";
```
Comment (figsoda) [inline pkgs/tools/misc/fetch-scm/default.nix:39]: either option should work
```suggestion
      homepage = "https://github.com/KikyTokamuro/fetch.scm";
```
```suggestion
      inherit (src.meta) homepage;
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/fetch-scm/default.nix (hunk 1 above)
Comment (q60) [reply pkgs/tools/misc/fetch-scm/default.nix]: done

File: pkgs/tools/misc/fetch-scm/default.nix (hunk 3 above)
Comment (q60) [reply pkgs/tools/misc/fetch-scm/default.nix]: done

File: pkgs/tools/misc/fetch-scm/default.nix (hunk 5 above)
Comment (q60) [reply pkgs/tools/misc/fetch-scm/default.nix]: removed it, will make new release later

File: pkgs/tools/misc/fetch-scm/default.nix (hunk 6 above)
Comment (q60) [reply pkgs/tools/misc/fetch-scm/default.nix]: not the best idea, rather unusual name for an application. also, i maintain package with the same name `fetch-scm` in AUR, it has `fetch-scm` exec

Comment (figsoda) [reply pkgs/tools/misc/fetch-scm/default.nix]: I think `fetch.scm` is a fine name, but fair enough

**Review by figsoda — APPROVED** [review]
there is a bug with `os-release-name`'s regex and it doesn't work on NixOS because of the parenthesis in `PRETTY_NAME`, but it should work fine on other distros

**Pull request merged**